### PR TITLE
inode/i_private: remove all unnecessary cast for i_private

### PR DESCRIPTION
**Pull request by anchao**
## Summary

inode/i_private: remove all unnecessary cast for i_private

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

File: drivers/sensors/hdc1008.c [hunk 1]
```diff
@@ -741,7 +741,7 @@ static int hdc1008_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 {
   FAR struct inode *inode        = filep->f_inode;
   FAR struct hdc1008_dev_s *priv =
-    (FAR struct hdc1008_dev_s *)inode->i_private;
+    inode->i_private;
```
Comment (xiaoxiang781216) [inline drivers/sensors/hdc1008.c:744]: merge to previous line

File: drivers/sensors/hdc1008.c [hunk 2]
```diff
@@ -593,7 +593,7 @@ static ssize_t hdc1008_read(FAR struct file *filep, FAR char *buffer,
...
```
Comment (xiaoxiang781216) [inline drivers/sensors/hdc1008.c:596]: merge to previous line

File: drivers/sensors/hdc1008.c [hunk 3]
```diff
@@ -518,7 +518,7 @@ static int hdc1008_open(FAR struct file *filep)
...
```
Comment (xiaoxiang781216) [inline drivers/sensors/hdc1008.c:521]: merge to previous line

File: drivers/sensors/hdc1008.c (hunk 1 above)
Comment (anchao) [reply drivers/sensors/hdc1008.c]: Done

File: drivers/sensors/hdc1008.c [hunk 4]
```diff
@@ -550,7 +550,7 @@ static int hdc1008_close(FAR struct file *filep)
...
```
[the remark this answers is not in the document]
Comment (anchao) [reply drivers/sensors/hdc1008.c]: Done

File: drivers/note/noteram_driver.c [hunk 5]
```diff
@@ -470,7 +470,7 @@ static int noteram_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 {
   int ret = -ENOSYS;
   FAR struct noteram_driver_s *drv =
-    (FAR struct noteram_driver_s *)filep->f_inode->i_private;
+    filep->f_inode->i_private;
```
[the remark this answers is not in the document]
Comment (anchao) [reply drivers/note/noteram_driver.c]: Done

File: drivers/sensors/hdc1008.c (hunk 3 above)
Comment (anchao) [reply drivers/sensors/hdc1008.c]: Done

File: drivers/sensors/hdc1008.c (hunk 2 above)
Comment (anchao) [reply drivers/sensors/hdc1008.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**